diff --git a/hw/arm/Kconfig b/hw/arm/Kconfig
index 82e0bc2..a2937b4 100644
--- a/hw/arm/Kconfig
+++ b/hw/arm/Kconfig
@@ -65,6 +65,16 @@ config EXYNOS4
     select USB_EHCI_SYSBUS
     select OR_IRQ
 
+config HALO
+    bool
+    default y
+    depends on TCG && ARM
+    select ARM_V7M
+    select SERIAL_MM
+    select UNIMP
+    select DESIGNWARE_I2C
+    select PL330
+
 config INTEGRATOR
     bool
     default y
diff --git a/hw/arm/meson.build b/hw/arm/meson.build
index 4233a80..b6e0f72 100644
--- a/hw/arm/meson.build
+++ b/hw/arm/meson.build
@@ -3,6 +3,22 @@ arm_common_ss.add(when: 'CONFIG_ARM_VIRT', if_true: files('virt.c'))
 arm_common_ss.add(when: 'CONFIG_ACPI', if_true: files('virt-acpi-build.c'))
 arm_common_ss.add(when: 'CONFIG_DIGIC', if_true: files('digic_boards.c'))
 arm_common_ss.add(when: 'CONFIG_EMCRAFT_SF2', if_true: files('msf2-som.c'))
+arm_common_ss.add(when: 'CONFIG_HALO', if_true: files(
+  'halo.c',
+  'halo_se.c',
+  'halo_wdog.c',
+  'halo_rtc.c',
+  'halo_adc.c',
+  'halo_utimer.c',
+  'halo_gpio.c',
+  'halo_ble.c',
+  'halo_cdc200.c',
+  'halo_dsi.c',
+  'halo_i2c_regfile.c',
+  'halo_wav.c',
+  'halo_i2s.c',
+  'halo_pdm.c',
+))
 arm_common_ss.add(when: 'CONFIG_INTEGRATOR', if_true: files('integratorcp.c'))
 arm_common_ss.add(when: 'CONFIG_MICROBIT', if_true: files('microbit.c'))
 arm_common_ss.add(when: 'CONFIG_MPS3R', if_true: files('mps3r.c'))
